flight/Modules/VibrationAnalysis/vibrationanalysis.c

```diff
@@ -83,17 +84,54 @@ static struct VibrationAnalysis_data {
 	float accels_static_bias_y; // (0,0,-g). In the case where they are not, this will still  
 	float accels_static_bias_z; // converge to the true bias in a few thousand measurements.	
 	
-	int16_t *accel_buffer_complex_x_q15;
-	int16_t *accel_buffer_complex_y_q15;
-	int16_t *accel_buffer_complex_z_q15;
-	
-	int16_t *fft_output;
+	int16_t *accel_buffer_x;
+	int16_t *accel_buffer_y;
+	int16_t *accel_buffer_z;
 } *vtd;
 
 
 // Private functions
 static void VibrationAnalysisTask(void *parameters);
 
+/*
+*   Releases any memory dinamically allocated
+*   Because this module can have a big footprint this will ensure we can
+*   reduce it when something fails.
```

Comment:
That could work, but it could also introduce some artifacts. I have this running on the simulator an can compare the two approaches.